.github/workflows/android_tests.yml

```diff
@@ -46,5 +48,18 @@ jobs:
       - name: Lint check
         run: ./gradlew ktlintCheck
 
-      - name: Execute unit tests
-        run: ./gradlew testDebugUnitTest
+      - name: Unit tests with code coverage reporting
+        run: ./gradlew jacocoTestReport
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: ./gradlew sonar
```

Comment:
Not sure if this is good idea - this will publish new sonar report with only android coverage and lint errors?